metrics/statusMetrics.go

```diff
@@ -0,0 +1,89 @@
+package metrics
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-proxy-go/data"
+)
+
+// statusMetrics will handle displaying at /status/metrics all collected metrics
+type statusMetrics struct {
+	endpointMetrics        map[string]*data.EndpointMetrics
+	mutEndpointsOperations sync.RWMutex
+}
+
+// NewStatusMetrics will return an instance of the struct
+func NewStatusMetrics() *statusMetrics {
+	return &statusMetrics{
+		endpointMetrics: make(map[string]*data.EndpointMetrics),
+	}
+}
+
+// AddRequestData will add the received data to the metrics map
+func (sm *statusMetrics) AddRequestData(path string, withError bool, duration time.Duration) {
+	// TODO: analyze possible way of improving this function - launch on goroutines (with a max num of goroutines),
```

Comment:
as discussed, can be done with a buffered chan and a processing go routine

Reply:
changed TODO